nemo/collections/nlp/data/token_classification/punctuation_capitalization_dataset.py

```diff
@@ -1170,6 +1191,96 @@ def _read_dataset(
         text_lines, punct_labels_lines, capit_labels_lines = zip(*dataset)
         return text_lines, punct_labels_lines, capit_labels_lines, punct_unique_labels, capit_unique_labels
 
+    @staticmethod
+    def calc_batch_seq_length(queries: List[np.ndarray], length_is_multiple_of: int) -> int:
+        return ceil(max([len(elem) for elem in queries]) / length_is_multiple_of) * length_is_multiple_of
+
+    def _adjust_number_of_batches(
+        self,
+        input_ids: List[np.ndarray],
+        batch_beginnings: List[int],
+        batch_sizes: List[int],
+        batch_seq_lengths: List[int],
+    ) -> Tuple[List[int], List[int], List[int]]:
+        """
+        If length of ``batch_sizes`` list is not divisible by ``self.number_of_batches_is_multiple_of``, then
+        one or several batches are split into parts until number of batches is divisible by
+        ``self.number_of_batches_is_multiple_of``.
+
+        The method selects a batch and tries to slice smaller batches with 8 elements each from the batch. If
+        the batch cannot be sliced any further and there are still not enough batches, then the next batch from dataset
+        is selected.
+
+        If slicing batches of size 8 is not enough, then batches of size 1 are created.
+
+        If dataset is too small to create enough batches, then a warning is shown.
+
+        Args:
+            input_ids: tokenized queries of the dataset. `input_ids` are expected to be sorted by length in ascending
+                order.
+            batch_beginnings: indices of first elements of batches created inside :meth:`_mark_up_batches` method.
+                Expected to be sorted in ascending order.
+            batch_sizes: sizes of batches created inside :meth:`_mark_up_batches` method.
+            batch_seq_lengths: lengths of elements in batch after padding created inside :meth:`_mark_up_batches`
+                method.
+
+        Returns:
+            batch_beginnings: a list of indices in ``input_ids`` of first samples of every batch
+            batch_sizes: a list of numbers of samples in batches
+            batch_seq_lengths: a list of sequence lengths after padding for every batch
+        """
+        batch_beginnings, batch_sizes, batch_seq_lengths = batch_beginnings.copy(), batch_sizes.copy()
+        batch_seq_lengths = batch_seq_lengths.copy()
```

Comment:
Line 1232 here generated an error because there are three items on the left of the `=` sign and only two on the right. This can be fixed by either deleting `batch_seq_lengths` from the left side or adding `batch_seq_lengths.copy()` to the right side (and then deleting line 1233).

Reply:
Thank you! I broke this when fixed code style.